File: csharp-projects/Testeo/s092-ejercicios/p01.cpp
/*
    Implemente una funcion que reciba un array representado con un puntero a entero,
    asi como su longitud. La funcion debe mostrar en pantalla cada uno de los elementos
    del array y su direccion de memoria.
*/
#include <iostream>
using namespace std;

void printArray(int* array, int length){
    for(int i = 0; i < length; i++){
        cout << "Elemento " << i + 1 << ": " << *array << endl;
        cout << "Direccion de memoria" << array << endl;
        array++;
    }
}

int main(){
    int values[] = {6, 5, 4, 3, 2, 1, 0};
    printArray(values, sizeof(values) / sizeof(int));

}

File: csharp-projects/Testeo/s081-repaso/p01.hs
{-
Implemente una funcion que, dado un valor de n, genere una lista de la forma
    L = [1, -1, 1, -1, ....] (n elementos)
-}

main = do
    let list = [(-1) ^ x * (2 ^ x) | x <- [0..19]]
    print(list)
    print(generateList 10)

generateList n = [(-1) ^ x | x <- [0..(n-1)]]